File: Java/AllYourBaseAreBelongToUsException.java
package java.aybabtu;

/**
 * An {@code AllYourBaseAreBelongToUsException} is thrown when all your base are
 * belong to us.
 */
public class AllYourBaseAreBelongToUsException extends Exception {
    private static final long serialVersionUID = 2669454180887817304L;
    
    private static final String DEFAULT_MESSAGE =
            "All your base are belong to us";

    /**
     * Constructs a new exception with a default detail message.
     * The cause is not initialized, and may subsequently be initialized by a
     * call to {@link #initCause(Throwable)}.
     */
    public AllYourBaseAreBelongToUsException() {
        super(DEFAULT_MESSAGE);
    }

    /**
     * Constructs a new exception with the specified detail message.
     * The cause is not initialized, and may subsequently be initialized by a
     * call to {@link #initCause(Throwable)}.
     *
     * @param   message
     *          The detail message.
     *          If {@code null} is specified, a default detail message will be
     *          used.
     *          The detail message is saved for later retrieval by the
     *          {@link #getMessage()} method.
     */
    public AllYourBaseAreBelongToUsException(String message) {
        super(message == null ? DEFAULT_MESSAGE : message);
    }

    /**
     * Constructs a new exception with the specified detail message and cause.
     *
     * <p>Note that the detail message associated with {@code cause} is
     * <em>not</em> automatically incorporated in this exception's detail
     * message.
     *
     * @param   message
     *          The detail message.
     *          If {@code null} is specified, a default detail message will be
     *          used.
     *          The detail message is saved for later retrieval by the
     *          {@link #getMessage()} method.
     * @param   cause
     *          The cause, which is saved for later retrieval by the
     *          {@link #getCause()} method.
     *          A {@code null} value is permitted, and indicates that the cause
     *          is nonexistent or unknown.
     */
    public AllYourBaseAreBelongToUsException(String message, Throwable cause) {
        super(message == null ? DEFAULT_MESSAGE : message, cause);
    }

    /**
     * Constructs a new exception with the specified cause and a detail message
     * of {@code cause.toString()} (which typically contains the class and
     * detail message of {@code cause}), or a default detail message if
     * {@code cause} is {@code null}.
     *
     * @param   cause
     *          The cause, which is saved for later retrieval by the
     *          {@link #getCause()} method.
     *          A {@code null} value is permitted, and indicates that the cause
     *          is nonexistent or unknown: in this case, a default detail
     *          message will be used.
     */
    public AllYourBaseAreBelongToUsException(Throwable cause) {
        super(cause == null ? DEFAULT_MESSAGE : cause.toString(), cause);
    }
}

File: C#/AllYourBaseAreBelongToUsException.cs
using System;
using System.Runtime.InteropServices;
using System.Runtime.Serialization;

namespace System
{
    // Summary:
    //      The exception that is thrown when all yor base are belong to us.
    [Serializable]
    [ComVisible(true)]
    public class AllYourBaseAreBelongToUsException : Exception
    {
        private const String DefaultMessage = "All your base are belong to us";

        // Summary:
        //      Initializes a new instance of the System.AYBABTU.AllYourBaseAreBelongToUsException
        //      class.
        public AllYourBaseAreBelongToUsException() : base(DefaultMessage)
        { }

        // Summary:
        //      Initializes a new instance of the System.AYBABTU.AllYourBaseAreBelongToUsException
        //      class with a specified error message.
        //
        // Parameters:
        //  message:
        //     The message that describes the error.
        public AllYourBaseAreBelongToUsException(string message) :
            base(message == null ? DefaultMessage : message)
        { }

        // Summary:
        //      Initializes a new instance of the System.AYBABTU.AllYourBaseAreBelongToUsException
        //      class with serialized data.
        //
        // Parameters:
        //  info:
        //      The object that holds the serialized object data.
        //
        //  context:
        //      The contextual information about the source or destination.
        protected AllYourBaseAreBelongToUsException(
            SerializationInfo info,
            StreamingContext context
            ) : base(info, context)
        { }

        // Summary:
        //      Initializes a new instance of the System.AYBABTU.AllYourBaseAreBelongToUsException
        //      class with a specified error message and a reference to the inner exception that is
        //      the cause of this exception.
        //
        // Parameters:
        //  message:
        //      The error message that explains the reason for the exception.
        //
        //  innerException:
        //      The exception that is the cause of the current exception.
        //      If the innerException parameter is not a null reference, the current exception is
        //      raised in a catch block that handles the inner exception.
        public AllYourBaseAreBelongToUsException(string message, Exception innerException) :
            base(message == null ? DefaultMessage : message, innerException)
        { }
    }
}
